pebble_tool/util/browser.py

```diff
@@ -47,9 +56,74 @@ def url_append_params(self, url, params):
+                               'static/js/underscore-min.js',
+                               'static/js/underscore-min.map',
+                               'static/js/websocket.js',
+                               'static/compass-rose.png',
+                               'static/compass-needle.png',
+                               'static/stylesheets/sensors.css']
+
+            def do_HEAD(self):
+                self.send_response(200)
+                self.end_headers()
+
+            def do_GET(self):
+                requested_file = self.path.rsplit('/', 1)[1]
+                file_path = self.path.lstrip('/')
+                if requested_file == '':
+                    self.send_response(200)
+                    self.send_header('Content-type', 'text/html')
+                    self.end_headers()
+                    self.wfile.write(SENSOR_PAGE_HTML.format(websocket_host="'{}'".format(gethostbyname(gethostname())),
+                                                             websocket_port="'{}'".format(pypkjs_port[0])))
+                elif file_path in self.PERMITTED_PATHS:
+                    try:
+                        file_contents = open(os.path.join(os.path.dirname(os.path.realpath(__file__)), file_path))
+                        self.send_response(200)
+                        self.end_headers()
+                        self.wfile.write(file_contents.read())
+                    except IOError:
+                        self.send_response(404)
+                        self.end_headers()
+                        self.wfile.write("Not Found")
+                else:
+                    self.send_response(403)
+                    self.end_headers()
+                    self.wfile.write("Forbidden")
+
+            def log_request(self, code='-', size='-'):
+                logger.debug("{} - - [{}] '{}' {} {}".format(self.client_address[0], self.log_date_time_string(),
+                                                               self.requestline, code, size))
+
+        server = BaseHTTPServer.HTTPServer(('', port), SensorPageHandler)
+        url = "{}:{}".format(gethostbyname(gethostname()), server.server_port)
+        url_code = pyqrcode.create(url)
+        print(url_code.terminal(quiet_zone=1))
+        print("=======================================================================================================")
+        print("Please scan the QR code or enter the following URL in your mobile browser:\n{}".format(url))
+        print("=======================================================================================================")
+        print("\nUse Ctrl-C to stop sending sensor data to the emulator.\n")
+        try:
+            server.serve_forever()
+        except KeyboardInterrupt:
+            print("Stopping...")
+            server.server_close()
+            time.sleep(2) # Wait for WS connection to die between phone/QEMU
+
+    def _choose_port(self, port=None):
```

Comment:
If `port` is not None, this function should probably just not be called. Otherwise this is just the identity function